tests/testthat/helper-shinytest2.R

```diff
@@ -0,0 +1,115 @@
+library(shinytest2)
+library(rvest)
+
+default_idle_timeout <- 20000
+
+
+default_idle_timeout <- 20000
+
+simple_teal_data <- function() {
+  data <- within(teal_data(), {
+    iris <- iris
+    mtcars <- mtcars
+  })
+  datanames(data) <- c("iris", "mtcars")
+  data
+}
+
+report_module <- function(label = "example teal module") {
+  module(
+    label = label,
+    server = function(id, data, reporter) {
+      moduleServer(id, function(input, output, session) {
+        teal.reporter::simple_reporter_srv(
+          id = "reporter",
+          reporter = reporter,
+          card_fun = function(card) card
+        )
+        updateSelectInput(session, "dataname", choices = isolate(datanames(data())))
+        output$dataset <- renderPrint({
+          req(input$dataname)
+          data()[[input$dataname]]
+        })
+      })
+    },
+    ui = function(id) {
+      ns <- NS(id)
+      sidebarLayout(
+        sidebarPanel(
+          teal.reporter::simple_reporter_ui(ns("reporter")),
+          selectInput(ns("dataname"), "Choose a dataset", choices = NULL)
+        ),
+        mainPanel(verbatimTextOutput(ns("dataset")))
+      )
+    }
+  )
+}
+
+#' Get all the module tabs in the `AppDriver` object along with their
+#' root_ids and module_ids used for name spacing in the shiny ui.
+get_app_module_tabs <- function(app) {
+  lapply(
+    app$get_html(selector = "ul.shiny-bound-input"),
```

Comment:
I'm concerned that this is too generic. Should make the selector more specific

```suggestion
    app$get_html(selector = "ul.nav.shiny-tab-input.shiny-bound-input"),
```

Or search for `ul` with id that ends on `active_tab`

```suggestion
    app$get_html(selector = "ul.shiny-bound-input[id$='active_tab']")
```